metakb/schemas.py

```diff
@@ -0,0 +1,274 @@
+    """Define constraints for predictive predicate."""
+
+    SENSITIVITY = 'predicts_sensitivity_to'
+    RESISTANCE = 'predicts_resistance_to'
+    REDUCED_SENSITIVITY = 'predicts_reduced_sensitivity_to'
+    ADVERSE_RESPONSE = 'predicts_adverse_response_to'
+
+
+# TODO: Check and change values for predicates
+class DiagnosticPredicate(str, Enum):
+    """Define constraints for diagnostic predicate."""
+
+    POSITIVE = 'diagnoses_positive_to'
+    NEGATIVE = 'diagnoses_negative_to'
+
+
+class PrognosticPredicate(str, Enum):
+    """Define constraints for prognostic predicate."""
+
+    BETTER_OUTCOME = 'prognoses_better_outcome_to'
+    POOR_OUTCOME = 'prognoses_poor_outcome_to'
+    POSITIVE = 'prognoses_positive_to'
+
+
+class PredisposingPredicate(str, Enum):
+    """Define constraints for predisposing predicate."""
+
+    POSITIVE = 'predisposes_positive_to'
+    UNCERTAIN_SIGNIFICANCE = 'predisposes_uncertain_significance_to'
+    LIKELY_PATHOGENIC = 'predisposes_likely_pathogenic_to'
+    PATHOGENIC = 'predisposes_pathogenic_to'
+
+
+class FunctionalPredicate(str, Enum):
+    """Define constraints for functional predicate."""
+
+    GAIN_OF_FUNCTION = 'functions_gain_to'
+    LOSS_OF_FUNCTION = 'functions_loss_to'
+    UNALTERED_FUNCTION = 'functions_unaltered_to'
+    NEOMORPHIC = 'functions_neomorphic_to'
+    DOMINATE_NEGATIVE = 'functions_dominate_negative_to'
+    UNKNOWN = 'functions_unknown_to'
+
+
+class VariationOrigin(str, Enum):
+    """Define constraints for variant origin."""
+
+    SOMATIC = 'somatic'
+    RARE_GERMLINE = 'rare_germline'
+    COMMON_GERMLINE = 'common_germline'
+    UNKNOWN = 'unknown'
+    NOT_APPLICABLE = 'N/A'
+
+
+class Direction(str, Enum):
+    """Define constraints for evidence direction."""
+
+    SUPPORTS = 'supports'
+    DOES_NOT_SUPPORT = 'does_not_support'
+
+
+class MoleculeContext(str, Enum):
+    """Define constraints for types of molecule context."""
+
+    GENOMIC = 'genomic'
+    TRANSCRIPT = 'transcript'
+    PROTEIN = 'protein'
+
+
+class Gene(BaseModel):
+    """GA4GH Gene Value Object."""
+
+    gene_id: str
+    type = "Gene"
+
+
+class Disease(BaseModel):
+    """Disease Value Object"""
+
+    disease_id: str
+    type = "Disease"
+
+
+class Therapy(BaseModel):
+    """Therapy Value Object"""
+
+    therapy_id: str
+    type = "Therapy"
+
+
+class TherapeuticResponseProposition(BaseModel):
+    """Define therapeutic Response Proposition model"""
+
+    id: str = Field(..., alias='_id')
+    type = PropositionType.PREDICTIVE.value
+    predicate: Optional[PredictivePredicate]
+    variation_origin: Optional[VariationOrigin]
+    has_originating_context: str  # vrs:Variation
+    disease_context: str  # vicc:Disease
+    therapy: str  # Therapy value object
+
+
+class AssertionMethodID(IntEnum):
+    """Create constraints for AssertionMethod ids for harvested sources."""
+
+    CIVIC_EID_SOP = 1
+    CIVIC_AID_AMP_ASCO_CAP = 2
+    CIVIC_AID_ACMG = 3
```

Comment:
This is for the `AssertionMethod.id` . Originally, I hardcoded the `id` for CIViC `assertion_methods`. @jsstevenson pointed out that this would cause duplicates once MOA and other sources are added. As of now, we're using these constants across sources we harvest to ensure uniqueness. Let me know if you think there's a better way to do this!